src/Dialect/ONNX/MLIRDialectBuilder.cpp

```diff
@@ -48,28 +54,71 @@ Value MathBuilder::exp(Value val) {
   return b.create<math::ExpOp>(loc, val);
 }
 
+Value MathBuilder::exp2(Value val) {
+  assert(val.getType().isa<FloatType>() && "Data type must be float.");
+  return b.create<math::Exp2Op>(loc, val);
+}
+
+Value MathBuilder::log2(Value val) {
+  assert(val.getType().isa<FloatType>() && "Data type must be float.");
+  return b.create<math::Log2Op>(loc, val);
+}
+
 Value MathBuilder::sgt(Value lhs, Value rhs) {
   if (lhs.getType().isa<IndexType, IntegerType>() ||
       lhs.getType().isa<IndexType>())
     return b.create<CmpIOp>(loc, CmpIPredicate::sgt, lhs, rhs);
   return b.create<CmpFOp>(loc, CmpFPredicate::OGT, lhs, rhs);
 }
+
+Value MathBuilder::sge(Value lhs, Value rhs) {
+  if (lhs.getType().isa<IndexType, IntegerType>() ||
+      lhs.getType().isa<IndexType>())
+    return b.create<CmpIOp>(loc, CmpIPredicate::sge, lhs, rhs);
+  return b.create<CmpFOp>(loc, CmpFPredicate::OGE, lhs, rhs);
+}
+
 Value MathBuilder::slt(Value lhs, Value rhs) {
   if (lhs.getType().isa<IndexType, IntegerType>() ||
       lhs.getType().isa<IndexType>())
     return b.create<CmpIOp>(loc, CmpIPredicate::slt, lhs, rhs);
   return b.create<CmpFOp>(loc, CmpFPredicate::OLT, lhs, rhs);
 }
+
 Value MathBuilder::eq(Value lhs, Value rhs) {
   if (lhs.getType().isa<IndexType, IntegerType>() ||
       lhs.getType().isa<IndexType>())
     return b.create<CmpIOp>(loc, CmpIPredicate::eq, lhs, rhs);
   return b.create<CmpFOp>(loc, CmpFPredicate::OEQ, lhs, rhs);
 }
+
 Value MathBuilder::select(Value cmp, Value lhs, Value rhs) {
   return b.create<SelectOp>(loc, cmp, lhs, rhs);
 }
 
+Value MathBuilder::constant(Type type, double val) {
```

Comment:
Thanks, I like that. I wonder if it might not make sense to make 2 calls, one for int and one for double, since they can be overloaded... Loved the call, but always feel a bit uneasy about casting an int to a double and expecting the right value... basically asking if `i == (int64_t)((double) i)` is true for any `i`.